editor/plugins/game_view_plugin.cpp

```diff
@@ -438,21 +686,86 @@ GameView::GameView(Ref<GameViewDebugger> p_debugger) {
 	menu->set_item_checked(menu->get_item_index(CAMERA_MODE_INGAME), true);
 	menu->add_radio_check_item(TTR("Manipulate From Editors"), CAMERA_MODE_EDITORS);
 
-	_update_debugger_buttons();
+	embedding_separator = memnew(VSeparator);
+	main_menu_hbox->add_child(embedding_separator);
+
+	keep_aspect_button = memnew(Button);
+	main_menu_hbox->add_child(keep_aspect_button);
+	keep_aspect_button->set_toggle_mode(true);
+	keep_aspect_button->set_theme_type_variation("FlatButton");
+	keep_aspect_button->set_tooltip_text(TTR("Keep the aspect ratio of the embedded game."));
+	keep_aspect_button->connect(SceneStringName(pressed), callable_mp(this, &GameView::_keep_aspect_button_pressed));
+
+	embed_options_menu = memnew(MenuButton);
+	main_menu_hbox->add_child(embed_options_menu);
+	embed_options_menu->set_flat(false);
+	embed_options_menu->set_theme_type_variation("FlatMenuButton");
+	embed_options_menu->set_h_size_flags(SIZE_SHRINK_END);
+	embed_options_menu->set_tooltip_text(TTR("Embedding Options"));
+
+	menu = embed_options_menu->get_popup();
+	menu->connect(SceneStringName(id_pressed), callable_mp(this, &GameView::_embed_options_menu_menu_id_pressed));
+	menu->add_check_item(TTR("Embed game on Next Play"), EMBED_RUN_GAME_EMBEDDED);
+	menu->add_check_item(TTR("Make Game Workspace floating on Next Play"), EMBED_MAKE_FLOATING_ON_PLAY);
+
+	main_menu_hbox->add_spacer();
+
+	game_size_label = memnew(Label());
+	main_menu_hbox->add_child(game_size_label);
+	game_size_label->hide();
 
 	panel = memnew(Panel);
 	add_child(panel);
 	panel->set_theme_type_variation("GamePanel");
 	panel->set_v_size_flags(SIZE_EXPAND_FILL);
 
+	embedded_process = memnew(EmbeddedProcess);
+	panel->add_child(embedded_process);
+	embedded_process->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+	embedded_process->connect(SNAME("embedding_failed"), callable_mp(this, &GameView::_embedding_failed));
+	embedded_process->connect(SNAME("embedding_completed"), callable_mp(this, &GameView::_embedding_completed));
+	embedded_process->connect(SNAME("embedded_process_updated"), callable_mp(this, &GameView::_embedded_process_updated));
```

Comment:
```suggestion
	embedded_process->connect("embedding_failed", callable_mp(this, &GameView::_embedding_failed));
	embedded_process->connect("embedding_completed", callable_mp(this, &GameView::_embedding_completed));
	embedded_process->connect("embedded_process_updated", callable_mp(this, &GameView::_embedded_process_updated));
```
The constructor is ran only once, so SNAME makes no sense here.

Reply:
Done!